overlord/snapstate/flags.go

```diff
@@ -50,3 +54,30 @@ func (f Flags) ForSnapSetup() Flags {
 	f.IgnoreValidation = false
 	return f
 }
+
+// EffectiveConfinement return the effective confinement for a given declared
+// confinement and flags like jailmode or devmode.
+func (f Flags) EffectiveConfinement(confinementType snap.ConfinementType) snap.ConfinementType {
+	switch confinementType {
+	case snap.ClassicConfinement:
+		// unchanged, flags don't impact snaps with classic confinement
+	case snap.DevmodeConfinement:
+		if f.JailMode {
+			// jailmode overrides devmode confinement declaration
+			return snap.StrictConfinement
+		}
+		// unchanged, stays devmode
+	case snap.StrictConfinement:
+		if f.JailMode {
+			// jailmode flag overrides devmode flag
+			return snap.StrictConfinement
+		}
```

Comment:
At a minimum, the comment here is not right and is confusing, but does this f.JailMode check make sense within the context of snap.StrictConfinement or can it just be removed?